quantecon/game_theory/normal_form_game.py

```diff
@@ -258,7 +258,7 @@ def reduce_last_player(payoff_array, action):
             if isinstance(action, numbers.Integral):  # pure action
                 return payoff_array.take(action, axis=-1)
             else:  # mixed action
-                return payoff_array.dot(action)
+                return payoff_array @ action
```

Comment:
I am not sure whether there would be no side effect from this change, when the dimension of `payoff_array` > 2.

Reply:
Many thanks @oyamad,

I think this is a really nice point as the equivalence is only established in 2-D arrays. Although all the tests are passed at the moment, it would be great to be cautious. In cases where `N-D.dot(1-D)`, the behaviour might also be unexpected.

I can write some tests on this while trying edge cases if you think it is a good idea.

Many thanks in advance.